source/codegen/templates/service.cpp.mako

```diff
@@ -16,6 +16,10 @@ function_names = service_helpers.filter_proto_rpc_functions_to_generate(function
 # If there are any non-mockable functions, we need to call the library directly, which
 # means we need another include file
 any_non_mockable_functions = any([not common_helpers.can_mock_function(functions[name]['parameters']) for name in function_names])
+# Define the constant for buffer too small if we have any of these functions.
+any_ivi_dance_functions = any(
+  [common_helpers.has_ivi_dance_with_a_twist_param(functions[name]['parameters']) or
```

Comment:
I think you could replace this with a generator comprehension (instead of list comprehension) to avoid evaluating the whole list.

Although DAQ will still be the worst case by far, unless/until we add ivi-dance-for-daq